mautic/mautic
# API - If entity exist, use PATCH on create action

…API controller



| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴🟢 
| New feature/enhancement? (use the a.x branch)      | 🔴🟢
| Deprecations?                          | 🔴🟢
| BC breaks? (use the c.x branch)        | 🔴🟢
| Automated tests included?              | 🔴🟢 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 



## Description

This PR change method to process data in API for create hook.
Before If entity exist, it update it, but clear all others data what was not send in payload due this lines https://github.com/mautic/mautic/blob/3c1ee87dddb9a9da0ca0058f92bad4809fa876de/app/bundles/ApiBundle/Controller/CommonApiController.php#L493








---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create API call POST /api/contacts/new

```
{
    "email": "example@example.com",
    "firstname": "name"
}
```
3. Create API call POST /api/contacts/new

```
{
    "email": "example@example.com"
}
```
4. See like firstname was removed 
5. After PR should works properly
